src/dmd/backend/cgelem.c

```diff
@@ -3576,10 +3576,13 @@ STATIC elem * eleq(elem *e, goal_t goal)
             e1->Eoper == OPvar &&
             (e2->Eoper == OPpair || e2->Eoper == OPrpair) &&
             goal == GOALnone &&
-            !el_appears(e2, e1->EV.sp.Vsym)
+            !el_appears(e2, e1->EV.sp.Vsym) &&
+            // Disable this rewrite if we're using x87 and the OPpair isn't made
+            // of FP values
```

Comment:
I think you mean OPpair isn't made of mixed FP and non-FP values?

Reply:
I reworded the comment a bit, I hope it's clearer now.